overlord/snapstate/snapstate.go

```diff
@@ -2145,6 +2260,64 @@ func doUpdate(ctx context.Context, st *state.State, names []string, updates []mi
 	return updated, updateTss, nil
 }
 
+type updateGroup struct {
+	infos []minimalInstallInfo
+	names []string
+}
+
+func splitEssentialUpdates(deviceCtx DeviceContext, updates []minimalInstallInfo, names []string) (essential, nonEssential *updateGroup) {
+	var snapd minimalInstallInfo
+	var bases []minimalInstallInfo
+	essential = &updateGroup{}
+	nonEssential = &updateGroup{}
+
+	addInfo := func(grp *updateGroup, info minimalInstallInfo) {
+		grp.infos = append(grp.infos, info)
+		if len(names) != 0 {
```

Comment:
Why checking `len(names)` is important? As this is the only way that `names` is used in the function, maybe making it a flag would make it more explanatory.

Reply:
Same as previous comment, this has been removed now. Thank for bringing this up